cloudify_agent/installer/__init__.py

```diff
@@ -387,9 +391,27 @@ def upload_certificate(self):
         cert_content = self.rest_cert_content
 
         if not self.runner.exists(cert_file):
+            self._create_cert_dir(cert_file)
             with tempfile.NamedTemporaryFile(delete=False) as f:
                 f.write(cert_content)
             try:
                 self.runner.put_file(f.name, cert_file)
             finally:
                 os.unlink(f.name)
+
+    def _create_cert_dir(self, cert_file):
+        """Create the directory containing the manager certificate.
+
+        For cross-platform compatibility, use python to create the directory.
+        """
+
+        try:
+            self.runner.python(
+                'import os',
+                'os.makedirs(os.path.dirname(os.path.expanduser(\'{0}\')))'
+                .format(cert_file))
+        except:
+            # an error was thrown - if the directory does exist, we assume
+            # it was a "directory already exists" error and continue.
+            if not self.runner.exists(cert_file):
```

Comment:
you should be checking the dirname which again leads to the question of behavior on windows.
i guess you can use python again if there is a problem